### ISSUE TITLE
TypeScript: "error" event specifies Error instead of ErrorEvent

### ISSUE THREAD

**Issue by alula** [abalabahaha/eris#944]
TypeScript typing for Eris specifies that "error" event emits an `Error`, but actually it emits an `ErrorEvent` object.

```
    client.on("error", (err, id) => {
        logger.error("An error occurred on shard " + id);
        logger.error(err.error); // Property 'error' does not exist on type 'Error'.ts(2339)
    });
```

output of code as above but second call is `logger.error(err);`
```
  23:42:20.667  ErisClient       ERROR    An error occurred on shard 0
  23:42:20.667  ErisClient       ERROR    ErrorEvent {
  target: WebSocket {
    _events: [Object: null prototype] {
      open: [Function],
      message: [Function],
      error: [Function]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    readyState: 2,
    protocol: '',
```

**Comment by bsian03** [comment]
From abal:
> I'd rather have the WS emit an `Error`, which I'm drawing up

**Comment by abalabahaha** [comment]
Fixed by d32cab8, which should land in a release soon
